### DISCUSSION TITLE
What is the best way to add compiler args with korge gradle plugin?

### DISCUSSION THREAD

**Discussion by BioRyajenka** [korlibs/korge#1457] [Q&A] [answered]
Hi, 
what is the best way to add compiler args with korge gradle plugin?

I am adding it like this 
```kotlin
tasks.withType(KotlinCompile::class).all {
    kotlinOptions.freeCompilerArgs += listOf("-Xcontext-receivers", "-Xskip-prerelease-check")
}
```
but both options seem to be ignored (I still see `Class '...' is compiled by a pre-release version of Kotlin and cannot be loaded by this version of the compiler`, for example)

**Comment by soywiz** [comment]
Maybe that's executeed synchronously at the moment the code is evaluated. And at that point targets might not have been created yet. Maybe a couple of afterEvaluate? Or maybe there is a way like .configureEeach that can be executed when the task type is created even if that happens later?

**Reply by BioRyajenka** [reply]
I don't see yet what I can do based on your suggestion but I will try to play around it.

I see you also use `-Xskip-prerelease-check` in a couple of places. Does it only affects korge repo? 
I mean, I see you added `-Xskip-prerelease-check` but in my project this argument doesn't seem to be applied. Is it expected?

**Comment by soywiz** [comment, marked as answer]
```kotlin
tasks.withType(KotlinCompile::class).all {
    kotlinOptions.freeCompilerArgs += listOf("-Xcontext-receivers", "-Xskip-prerelease-check")
}
```

should work fine. This was a temporal problem related to using a pre-release kotlin version.
